src/P/EitherT.hs

```diff
@@ -11,3 +13,6 @@ firstEitherT f = bimapEitherT f id
 
 secondEitherT :: (Functor m) => (a -> b) -> EitherT e m a -> EitherT e m b
 secondEitherT f = bimapEitherT id f
+
+eitherTFromMaybe :: Functor m => b -> m (Maybe a) -> EitherT b m a
+eitherTFromMaybe l r = EitherT $ maybe (Left l) pure <$> r
```

Comment:
Pulling in `lift` would require more dependencies in P, namely adding `transformers` to the short list of dependencies in `p.cabal` Not sure if that's ok?